app/controllers/api/v1/mc_questions_controller.rb

```diff
@@ -0,0 +1,67 @@
+class Api::V1::McQuestionsController < ApplicationController
+    before_action :set_mc_question, only: [:show, :edit, :update, :destroy]
+    skip_before_action :verify_authenticity_token
+
+    # GET /mc_questions or /mc_questions.json
+    def index
+      @mc_questions = McQuestion.all
+      render json: @mc_questions.to_json(include: :mc_options)
+    end
+  
+    # GET /mc_questions/1 or /mc_questions/1.json
+    def show
+        render json: @mc_question.to_json(include: :mc_options)
+    end
+  
+    # POST /mc_questions or /mc_questions.json
+    def create
+        @mc_question = McQuestion.new(params.require(:mc_question).permit(:question, :survey_id))
+        options = params.require(:mc_options).permit(:options => [])
+
+        options[:options].each do |option|
+          @mc_question.mc_options << McOption.new(option: option)
+        end
+
+        if @mc_question.save
+          render json: @mc_question
+        else
+          render json: {notice: "Failure! Could not save McQuestion due to #{@mc_question.errors.full_messages}"}
+        end
+    end
+  
+    # PATCH/PUT /mc_questions/1 or /mc_questions/1.json
+    def update
+      if @mc_question.survey.isLive == false
+        if @mc_question.update(params.require(:mc_question).permit(:question, :survey_id))
+
+          if @mc_question.mc_options.any?
+            @mc_question.mc_options.destroy_all
```

Comment:
Not sure about this, its the easiest way to update the options on the backend side of things since you don't have to make any assumptions about what options are new and send over a ton of id's for all the changed options. On the other hand it feels a little hacky.

Reply:
hmmm good points, but i think we can leave it, this code is easy to understand too